cl_dll/input.cpp

```diff
@@ -182,6 +125,74 @@ typedef struct kblist_s
 
 kblist_t *g_kbkeys = NULL;
 
+namespace autofuncs
+{
+	static cvar_t* cl_autojump;
+
+	static struct {
+		bool onground = false;
+		bool inwater = false;
+		bool walking = true; // Movetype == MOVETYPE_WALK. Filters out noclip, being on ladder, etc.
+	} player;
+
+	static void handle_autojump(usercmd_t* cmd)
+	{
+		static bool s_jump_was_down_last_frame = false;
+
+		if (cl_autojump->value != 0.0f)
+		{
+			bool should_release_jump = (!player.onground && !player.inwater && player.walking);
+
+			/*
+			 * Spam pressing and releasing jump if we're stuck in a spot where jumping still results in
+			 * being onground in the end of the frame. Without this check, +jump would remain held and
+			 * when the player exits this spot they would have to release and press the jump button to
+			 * start jumping again. This also helps with exiting water or ladder right onto the ground.
+			 */
+			if (s_jump_was_down_last_frame && player.onground && !player.inwater && player.walking)
+				should_release_jump = true;
+
+			if (should_release_jump)
+				cmd->buttons &= ~IN_JUMP;
+		}
+
+		s_jump_was_down_last_frame = ((cmd->buttons & IN_JUMP) != 0);
+	}
+
+	static void handle_ducktap(usercmd_t* cmd)
+	{
+		static bool s_duck_was_down_last_frame = false;
+		static bool duck_was_pressed = false;
+		static bool should_release_duck;
+
+		if (in_duck.state & 1)
```

Comment:
```suggestion
		static bool should_release_duck;

		bool duck_was_pressed = false;
		if (in_duck.state & 1)
```
Doesn't need to be a static if it's set every time.

Also please rename to `duck_is_pressed` and remove the set in the `should_release_duck` condition below as it does nothing.